sources/sqlserver/infoschema.go

```diff
@@ -0,0 +1,383 @@
+			continue
+		}
+		switch constraint {
+		case "PRIMARY KEY":
+			primaryKeys = append(primaryKeys, col)
+		default:
+			m[col] = append(m[col], constraint)
+		}
+	}
+	return primaryKeys, m, nil
+}
+
+// GetForeignKeys returns a list of all the foreign key constraints.
+func (isi InfoSchemaImpl) GetForeignKeys(conv *internal.Conv, table common.SchemaAndName) (foreignKeys []schema.ForeignKey, err error) {
+	q := `
+	SELECT 
+		OBJECT_SCHEMA_NAME (FK.referenced_object_id) AS [schema_name],
+		OBJECT_NAME (FK.referenced_object_id) AS [referenced_table],
+		COL_NAME(FKC.parent_object_id, FKC.parent_column_id) AS [column],  
+		COL_NAME(FKC.referenced_object_id, FKC.referenced_column_id) AS [referenced_column],  
+		FK.name AS [foreign_key_name]
+	FROM sys.foreign_keys AS FK  
+	INNER JOIN sys.foreign_key_columns AS FKC   
+    ON FK.object_id = FKC.constraint_object_id  
+	WHERE FK.parent_object_id = OBJECT_ID(@p1);
+	`
+
+	rows, err := isi.Db.Query(q, fmt.Sprintf("%s.%s", table.Schema, table.Name))
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+	var refTable common.SchemaAndName
+	var col, refCol, fKeyName string
+	fKeys := make(map[string]common.FkConstraint)
+	var keyNames []string
+	for rows.Next() {
+		err := rows.Scan(&refTable.Schema, &refTable.Name, &col, &refCol, &fKeyName)
+		if err != nil {
+			conv.Unexpected(fmt.Sprintf("Can't scan: %v", err))
+			continue
+		}
+		tableName := isi.GetTableName(refTable.Schema, refTable.Name)
+		if _, found := fKeys[fKeyName]; found {
+			fk := fKeys[fKeyName]
+			fk.Cols = append(fk.Cols, col)
+			fk.Refcols = append(fk.Refcols, refCol)
+			fKeys[fKeyName] = fk
+			continue
+		}
+		fKeys[fKeyName] = common.FkConstraint{Name: fKeyName, Table: tableName, Refcols: []string{refCol}, Cols: []string{col}}
+		keyNames = append(keyNames, fKeyName)
+	}
+
+	sort.Strings(keyNames)
+	for _, k := range keyNames {
+		foreignKeys = append(foreignKeys,
+			schema.ForeignKey{
+				Name:         fKeys[k].Name,
+				Columns:      fKeys[k].Cols,
+				ReferTable:   fKeys[k].Table,
+				ReferColumns: fKeys[k].Refcols})
+	}
+	return foreignKeys, nil
+}
+
+// GetIndexes return a list of all indexes for the specified table.
+func (isi InfoSchemaImpl) GetIndexes(conv *internal.Conv, table common.SchemaAndName) ([]schema.Index, error) {
+	q2 := `
+		SELECT
+			IX.name, 
+			COL_NAME(IX.object_id, IXC.column_id) as [Column Name],
+			IX.is_unique,
+			IXC.is_descending_key 
+		FROM sys.indexes IX 
+		INNER JOIN sys.index_columns IXC 
+			ON  IX.object_id = IXC.object_id AND IX.index_id = IXC.index_id
+		INNER JOIN sys.tables TAB 
+			ON IX.object_id = TAB.object_id 
+		WHERE
+			IX.is_primary_key = 0          
+			AND IX.is_unique_constraint = 0 
+			AND TAB.is_ms_shipped = 0   
+			AND TAB.name=@p1
+			AND TAB.schema_id = SCHEMA_ID(@p2)
+			ORDER BY IX.name ;
+	`
+	rows, err := isi.Db.Query(q2, table.Name, table.Schema)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+	// TODO : remove sequence
```

Comment:
What is this todo for?

Reply:
It was regarding maintaining the order of the column when fetching the index from the database, it is already completed, we forgot to remove the todo comment, will remove that in the next commit.